test/functionalTest/cases/2631_partial_datetimes/partial_datetime_creation.test

```diff
@@ -0,0 +1,91 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Support partial date time creations (only date, not time)
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. Creating date time using only date
+# 02. Check that it has been created corresponding to 00:00:00 that day
+#
+
+echo "01. Creating date time using only date"
+echo "======================================"
+payload='{
+  "id": "E",
+  "type": "T",
+  "D01": {
+    "value": "2016-10-01",
+    "type": "DateTime"
+  }
+}'
+orionCurl --url /v2/entities --payload "$payload"
+echo
+echo
+
+
+echo "02. Check that it has been created corresponding to 00:00:00 that day"
+echo "====================================================================="
+orionCurl --url /v2/entities/E
+echo
+echo
+
+
+--REGEXPECT--
+01. Creating date time using only date
+======================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/E?type=T
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+02. Check that it has been created corresponding to 00:00:00 that day
+=====================================================================
+HTTP/1.1 200 OK
+Content-Length: 95
+Content-Type: application/json
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+{
+    "D01": {
+        "metadata": {},
+        "type": "DateTime",
+        "value": "2016-10-01T00:00:00.00Z"
```

Comment:
To sum up:

Timezoneless datestrings refer to a local time, what we cannot assume to be Zulu. 

So my proposal is not to accept date strings without the timezone component to avoid errors. 

What do you think?